tests/test_resources/test_modules/test_folder.py

```diff
@@ -182,49 +163,35 @@ def test_cluster_and_gcs(self, cluster, cluster_folder):
                 f"on the cluster {cluster.name}. For now please manually enable them directly on the cluster. "
                 f"See https://cloud.google.com/sdk/gcloud/reference/auth/login"
             )
+        
+    @pytest.mark.level("minimal") # TODO: Needs S3 and GCP credentials
+    def test_folder_and_folder(self, folder):
+        # from one cloud folder to another cloud folder
+        assert "sample_file_0.txt" in folder.ls(full_paths=False)
 
-    @pytest.mark.level("minimal")  # Needs S3 credentials
-    def test_s3_and_s3(self, local_folder, s3_folder):
-        # from one s3 folder to another s3 folder
-        new_s3_folder = s3_folder.to(system="s3")
-        assert "sample_file_0.txt" in new_s3_folder.ls(full_paths=False)
-
-        new_s3_folder.rm()
-
-    @pytest.mark.level(
-        "minimal"
-    )  # TODO: fix this test (Needs google-cloud-storage package)
-    @pytest.mark.skip("[WIP] Fix this test")
-    def test_gcs_and_gcs(self, gcs_folder):
-        # from one gcs folder to another gcs folder
-        new_gcs_folder = gcs_folder.to(system="gs")
-        assert "sample_file_0.txt" in new_gcs_folder.ls(full_paths=False)
-
-        new_gcs_folder.rm()
+        folder.rm()
 
-    @pytest.mark.level("minimal")  # TODO: fix this test
+    @pytest.mark.level("minimal") # TODO: Fix this test 
     @pytest.mark.skip("Doesn't work properly as only full-bucket copy is supported")
-    def test_s3_and_gcs(self, s3_folder):
+    def test_cloudA_and_cloudB(self, folder): 
         # *** NOTE: transfers between providers are only supported at the bucket level at the moment (not directory) ***
 
-        s3_folder_to_gcs = s3_folder.to(system="gs")
-        assert "sample_file_0.txt" in s3_folder_to_gcs.ls(full_paths=False)
-        assert s3_folder_to_gcs._fs_str == "gs"
-
-        s3_folder_to_gcs.rm()
+        if folder._fs_str == "s3":  # test s3 and gcs
+            folder = folder.to(system="gs")
+            assert folder._fs_str == "gs"
+        elif folder._fs_str == "gs": # test gcs and s3
+            folder = folder.to(system="s3")
+            assert folder._fs_str == "s3"
+        elif folder._fs_str == "ssh": # test ssh and s3
+            folder = folder.to(system="s3")
+            assert folder._fs_str == "s3"
+        elif folder._fs_str == "file": # test local and s3
+            folder = folder.to(system="s3")
+            assert folder._fs_str == "s3"
```

Comment:
for each input folder, we should be testing sending it to all possible system options. e.g s3 -> s3, gs, azure, cluster, local to make sure that we cover all cases.

think about how you would go about parameterizing the dest system in a way that when this test is run on "local", it only runs the cluster to cluster + here case; and on "minimal" level, it runs all combinations.